[Bug]: Python 3.12 vLLM v0.10.2+CPU build incompatibility with blake3==1.07

### Your current environment

This will reproduce the failing build of vLLM (using Python 3.12) v0.10.2+CPU with blake3 v1.07
```console
nix-build -I nixpkgs=https://github.com/daniel-fahey/nixpkgs/archive/e7fb8960241ab2211c471004bcd2b04683767c33.tar.gz   --expr 'with import <nixpkgs> { }; python312Packages.vllm'
```

i.e. using this exact [Nixpkgs derivation](https://github.com/daniel-fahey/nixpkgs/blob/e7fb8960241ab2211c471004bcd2b04683767c33/pkgs/development/python-modules/vllm/default.nix). (A Nixpkgs derivation is a package build recipe that defines how to compile and package software with all its dependencies in the Nix package manager ecosystem.)

See my Nixbuild.net Build #4565380 for the failed build details and logs [here](https://nixbuild.net/builds/4565380?t=EtkBCm8KBWJ1aWxkCgpidWlsZDpyZWFkGAMiCQoHCAcSAxD5BSINCgsIBBIHOgUKAxiBCDImCiQKAggbEgYIBRICCAUaFgoECgIIBQoICgYg9eyp0wYKBBoCCAAyFgoUCgIIGxIOCAISAxiACBIFEITTlgISJAgAEiAGRu1AM51kzPYGl4nShP2nj4SwLZdIQEZhBaIO6KS15xpATQzlvJ7Y-SLutaWbRFkbSrm3qOrarrHr60v_1kS5vx08FGnOtkZSZjwNvMDFTJwiO1O-MfmWac0BNAXtrNz8ASIiCiAuWrCOa6uQpEanWP7s1LJ8iKWZhtugtJoeboULO7JD1w==).

### 🐛 Describe the bug

vLLM v0.10.2 exhibits a somewhat bizarre build incompatibility with Python 3.12 when paired with blake3==1.07 in CPU-only configurations.

This incompatibility surfaces during the build process where blake3 version 1.07 conflicts with vLLM's build system when targeting CPU-only deployments. As Nixpkgs maintainers, we've decided to mark this particular configuration as broken in Nixpkgs for now, but thought our accidental discovery might be of interest to you as vLLM developers.

**Environment Details:**
- Python version: 3.12
- vLLM version: 0.10.2
- blake3 version: 1.07
- Target device: CPU (VLLM_TARGET_DEVICE=cpu)
- Build system: CMake/Ninja via Python setuptools
- Platform: x86_64-linux

**Reproduction:**
The environment section above contains a Nix command that reliably reproduces this failure, along with detailed build logs demonstrating the specific incompatibility between these versions.

**Build Failure (most relevant lines from the Nixbuild.net log linked above):**
```console
[1/18] Building CXX object CMakeFiles/dnnl_ext.dir/csrc/cpu/dnnl_helper.cpp.o[KKK
FAILED: [code=1] CMakeFiles/dnnl_ext.dir/csrc/cpu/dnnl_helper.cpp.o 
/nix/store/95k9rsn1zsw1yvir8mj824ldhf90i4qw-gcc-wrapper-14.3.0/bin/g++ -DCPU_CAPABILITY_AVX512 -I/build/source/csrc -I/include -I/src -O2 -g -DNDEBUG -std=gnu++17 -mf16c -fopenmp -DVLLM_CPU_EXTENSION -mavx512f -mavx512vl -mavx512bw -mavx512dq -mavx512bf16 -mavx512vnni -fPIC -MD -MT CMakeFiles/dnnl_ext.dir/csrc/cpu/dnnl_helper.cpp.o -MF CMakeFiles/dnnl_ext.dir/csrc/cpu/dnnl_helper.cpp.o.d -o CMakeFiles/dnnl_ext.dir/csrc/cpu/dnnl_helper.cpp.o -c /build/source/csrc/cpu/dnnl_helper.cpp
/build/source/csrc/cpu/dnnl_helper.cpp:4:10: fatal error: common/memory_desc.hpp: No such file or directory
    4 | #include "common/memory_desc.hpp"
      |          ^~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
[15/18] Building CXX object CMakeFiles/_C.dir/csrc/cpu/attention.cpp.ocpp.o
ninja: build stopped: subcommand failed.
```

**Error Context:**
This issue was discovered whilst updating the NixOS package for vLLM (see NixOS/nixpkgs#447722) to include Python 3.13 support. During testing, we saw a regression where CPU build configurations with blake3==1.07 failed to complete successfully, which was unexpected given that CUDA builds using the same derivation succeeded.

**Expected Behaviour:**
vLLM should build successfully on Python 3.12 with blake3==1.07 and higher versions in CPU-only configurations without requiring CUDA dependencies or throwing build errors.

**Clarifying Remark:**
To be clear, we do have working Nixpkgs derivations for vLLM v0.10.2 Python 3.13 CPU and Python 3.13 CPU + CUDA configurations that are soon to be merged into our `master` branch.

### Before submitting a new issue...

- [x] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

--- a/cmake/cpu_extension.cmake
+++ b/cmake/cpu_extension.cmake
@@ -198,13 +198,24 @@ else()
 endif()
 
 if ((AVX512_FOUND AND NOT AVX512_DISABLED) OR (ASIMD_FOUND AND NOT APPLE_SILICON_FOUND) OR POWER9_FOUND OR POWER10_FOUND OR POWER11_FOUND)
-    FetchContent_Declare(
-        oneDNN
-        GIT_REPOSITORY https://github.com/oneapi-src/oneDNN.git
-        GIT_TAG v3.9
-        GIT_PROGRESS TRUE
-        GIT_SHALLOW TRUE
-    )
+    set(FETCHCONTENT_SOURCE_DIR_ONEDNN "$ENV{FETCHCONTENT_SOURCE_DIR_ONEDNN}" CACHE PATH "Path to a local oneDNN source directory.")
+
+    if(FETCHCONTENT_SOURCE_DIR_ONEDNN)
+        message(STATUS "Using oneDNN from specified source directory: ${FETCHCONTENT_SOURCE_DIR_ONEDNN}")
+        FetchContent_Declare(
+            oneDNN
+            SOURCE_DIR ${FETCHCONTENT_SOURCE_DIR_ONEDNN}
+        )
+    else()
+        message(STATUS "Downloading oneDNN from GitHub")
+        FetchContent_Declare(
+            oneDNN
+            GIT_REPOSITORY https://github.com/oneapi-src/oneDNN.git
+            GIT_TAG v3.9
+            GIT_PROGRESS TRUE
+            GIT_SHALLOW TRUE
+        )
+    endif()
 
     if(USE_ACL)
         find_library(ARM_COMPUTE_LIBRARY NAMES arm_compute PATHS $ENV{ACL_ROOT_DIR}/build/)
@@ -227,7 +238,7 @@ if ((AVX512_FOUND AND NOT AVX512_DISABLED) OR (ASIMD_FOUND AND NOT APPLE_SILICON
     set(ONEDNN_ENABLE_ITT_TASKS "OFF")
     set(ONEDNN_ENABLE_MAX_CPU_ISA "OFF")
     set(ONEDNN_ENABLE_CPU_ISA_HINTS "OFF")
-    set(ONEDNN_VERBOSE "ON")
+    set(ONEDNN_VERBOSE "OFF")
     set(CMAKE_POLICY_DEFAULT_CMP0077 NEW)
 
     FetchContent_MakeAvailable(oneDNN)

